components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2178,25 +2171,27 @@ private UsersGetResponse getMultiAttributeFilteredUsers(Node node, Map<String, B
             paginatedUserResult = getMultiAttributeFilteredUsersWithMaxLimit(node, offset, sortBy, sortOrder, domainName, limit,
                     true);
         } else {
-            paginatedUserResult = getMultiAttributeFilteredUsersWithMaxLimit(node, offset, sortBy, sortOrder, domainName, maxLimit,
-                    false);
+            int maxLimit = getMaxLimit(domainName);
+            paginatedUserResult = getMultiAttributeFilteredUsersWithMaxLimit(node, offset, sortBy, sortOrder,
+                    domainName, maxLimit, false);
```

Comment:
Fix the 2nd problem mentioned here https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/609#issuecomment-2644715275

When retrieving results when the count is not specified, consider the max limit defined in the corresponding user store domain.